repo/fsrepo/fsrepo.go

```diff
@@ -359,14 +359,22 @@ func (r *FSRepo) Path() string {
 
 // SetAPIAddr writes the API Addr to the /api file.
 func (r *FSRepo) SetAPIAddr(addr ma.Multiaddr) error {
-	f, err := os.Create(filepath.Join(r.path, apiFile))
+	// Create a temp file to write the address, so that we don't leave empty file when the
+	// program crashes after creating the file.
+	f, err := os.Create(filepath.Join(r.path, apiFile+"-tmp"))
```

Comment:
I'd name this something like:

```suggestion
	f, err := os.Create(filepath.Join(r.path, "." + apiFile+".tmp"))
```